### PR TITLE
Extract frozen cache infrastructure to separate module

### PR DESCRIPTION
**Pull request by original-brownbear**
This extracts all infrastructure for the frozen shared cache up to but not including the frozen index input which isn't directly reusable due to its use of the internal cache index. This doesn't (aside from that one test utils class that I split in two) change any code.
I did not yet change any setting names since that seemed secondary but I'd go for this step to set up further more technical steps that involve actual code changes.

### CODE REVIEW

**Comment by original-brownbear** [comment]
Jenkins run elasticsearch-ci/part-1

File: x-pack/plugin/blob-cache/src/main/java/org/elasticsearch/blobcache/BlobCachePlugin.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+package org.elasticsearch.blobcache;
+
+import org.elasticsearch.blobcache.shared.SharedBlobCacheService;
+import org.elasticsearch.common.settings.Setting;
+import org.elasticsearch.plugins.ExtensiblePlugin;
+import org.elasticsearch.plugins.Plugin;
+
+import java.util.List;
+
+public class BlobCachePlugin extends Plugin implements ExtensiblePlugin {
```
Comment (original-brownbear) [inline x-pack/plugin/blob-cache/src/main/java/org/elasticsearch/blobcache/BlobCachePlugin.java:16]: Making this `ExtensiblePlugin` and using it from `SearchableSnapshots` was the only way I found to make the classloading work out such that the `preallocate` permissions and the settings are guaranteed to get set up before using them in `SearchableSnapshots`.

File: x-pack/plugin/blob-cache/src/test/java/org/elasticsearch/blobcache/BlobCacheTestUtils.java [hunk 2]
```diff
@@ -0,0 +1,337 @@
...
+
+    public static long sumOfCompletedRangesLengths(CacheFile cacheFile) {
+        return cacheFile.getCompletedRanges().stream().mapToLong(ByteRange::length).sum();
+    }
+
+    /**
+     * A {@link BlobContainer} that can read a single in-memory blob.
+     * Any attempt to read a different blob will throw a {@link FileNotFoundException}
+     */
+    public static BlobContainer singleBlobContainer(final String blobName, final byte[] blobContent) {
```
Comment (tlrx) [inline x-pack/plugin/blob-cache/src/test/java/org/elasticsearch/blobcache/BlobCacheTestUtils.java:157]: I think we can leave this method (and also `singleSplitBlobContainer`)  in the searchable snapshots plugin

**Review by tlrx — APPROVED** [review]
LGTM

**Comment by original-brownbear** [comment]
Thanks Tanguy!

**Pull request merged**